### PR TITLE
[#12291] Admin logs page: Fix overflow of "Clear all" button

### LINKED ISSUE #12291: Admin logs page: Fix overflow of "Clear all" button
At screen sizes of <500px, the "Clear all" button overflows:
photo_2023-03-28 14 00 46

One possible fix is to place the button on a separate row altogether.

### PR DESCRIPTION
**Pull request by DeliaBarron**
Fixes #12291

**PR Checklist**

Ensure that you have:
- [x] Read and understood our PR guideline: https://teammates.github.io/teammates/process.html#step-4-submit-a-pr
  - [x] Added the issue number to the "Fixes" keyword above
  - [x] Titled the PR as specified in the above mentioned document
- [x] Made your changes on a branch other than `master` and `release`
- [x] Gone through all the changes in this PR and ensured that:
  - [x] They addressed one (and only one) issue
  - [x] No unintended changes were made
- [x] Run and passed static analysis: `./gradlew lint` and `npm run lint`
- [x] Added/updated tests, if changes in functionality were involved
- [ ] Added/updated documentation to public APIs (classes, methods, variables), if applicable

**Outline of Solution**

-- Displaying button in another row to give sense of its function (clearing all 4 inputs above). 
-- Changing the inputs display width to be a half of the row (col-6) space they're in to fit.

### CODE REVIEW

**Review by domlimm — CHANGES_REQUESTED** [review]
@DeliaBarron Changes look good! The `Clear all` button seems to clear all fields in the entire section from `Trace ID` to `Extra filters`, so moving the button to the bottom of the 4 fields you specified might not be ideal.

Also, ensure that all tests are passing. For the current ones that are failing, you could run `npm run lint` at the root of TEAMMATES to see which files are failing the check. If component tests are still failing after, please update the snapshot of the page (I assume they might fail after you fix the lint errors).

**Review by weiquu — CHANGES_REQUESTED** [review]
Hi @DeliaBarron, the layout of the input fields within the page also seems to have changed... I think let's try to keep them the same. Here's what I mean (your fix is above, original is below):

fix

original

btw the lint error seems to come from the change in the `.scss` file... not sure why there's a change there

**Comment by DeliaBarron** [comment]
Oh okok I'm on it!

**Comment by DeliaBarron** [comment]
Hello! @weiquu @domlimm
I did the changes on the button overflow but I just have a query...
Was it aright to hide the label text when to small of a screen? I suggest this quick style but not sure if hiding part of the label text will pass the standard. Like this.... (2nd is the suggestion)
Screen Shot 2023-03-29 at 14 46 23
Screen Shot 2023-03-29 at 14 46 36

Also, I keep getting this after running `npm run lint`:
```

grep: repetition-operator operand invalid
Please provide a list or glob pattern of files to lint.

```
I'm not getting errors anymore but this pops up and Ive been looking for some answers but Im not sure I should be modifying regex for this (cause is what I get as solutions for it)

Thanks!
I'll be tuned!

**Comment by weiquu** [comment]
> Was it aright to hide the label text when to small of a screen? I suggest this quick style but not sure if hiding part of the label text will pass the standard

I think let's not hide the label text, since we try to make the site as [mobile-friendly](https://teammates.github.io/teammates/best-practices/mobile-friendliness.html) as possible. Hiding the text might make it difficult for the user to know the purpose of the input field. I believe you are trying to fix the input fields being misaligned? We can look to other solutions, like re-aligning them even if the label text takes up multiple rows (e.g. in your first screenshot, shift the `Trace ID` input field down), or placing each input on a single row for smaller screen widths.

Also I noted that you're testing on screen widths that are smaller than 360px, which is smaller than the minimum screen width that we support (see link above).

Regarding the button itself, placing it on a separate row is fine, just not in between all the input fields that it is supposed to clear.

> Also, I keep getting this after running `npm run lint`:
> 
> ```
> 
> grep: repetition-operator operand invalid
> Please provide a list or glob pattern of files to lint.
> ```
> 
> I'm not getting errors anymore but this pops up and Ive been looking for some answers but Im not sure I should be modifying regex for this (cause is what I get as solutions for it)

No need to edit anything for the command! Just run the `npm run lint` and fix any lint errors that pop up (if any).

**Comment by zhaojj2209** [comment]
How about putting the "Clear all" button on the same row as the "Advanced filters" button?

**Comment by DeliaBarron** [comment]
Hello again! Sorry if Im making of this issue a more conflictive one than how it could be. Trying to learn to collaborate as fast as I can and satisfy the project's standards. Sorry if this has taken more time than expected again...
Screen Shot 2023-03-30 at 16 09 53
Screen Shot 2023-03-30 at 16 12 07
Screen Shot 2023-03-30 at 16 12 30

I think putting the button below the inputs makes more sense than putting it above. Like a send button showing what their own function covers. Let me know ill keep modifying in the meantime to see if I can find something better

File: src/web/app/pages-monitoring/logs-page/logs-page.component.html [hunk 1]
```diff
@@ -98,7 +98,9 @@ <h1>Logs</h1>
             <label for="source-file" class="fw-bold">Source location file</label>
             <input id="source-file" class="form-control col-12" [(ngModel)]="formModel.filters.sourceLocation!.file">
           </div>
-          <div class="col-12 mt-2">
+        </div>
+        <div class="col-sm-6 col-1">
+          <div class="col-12">
```
Comment (weiquu) [inline src/web/app/pages-monitoring/logs-page/logs-page.component.html:103]: I believe these lines (and the similar lines below) are causing the input fields to have a different layout, particularly for desktop view. I personally think we should keep the layout the same.

File: src/web/app/pages-monitoring/logs-page/logs-page.component.scss [hunk 2]
```diff
@@ -21,15 +21,15 @@
...
 
-  .col-4 {
+  .col-1 {
     width: auto;
   }
 
-  .col-1 {
+  .col-6 {
     width: auto;
   }
```
Comment (weiquu) [inline src/web/app/pages-monitoring/logs-page/logs-page.component.scss:34]: I think let's remove these lines so as not to override the Bootstrap default styles as the overriding might cause issues in the future. Sorry for any confusion caused! We must have missed these out when removing the other Bootstrap style overrides.

**Review by weiquu — CHANGES_REQUESTED** [review]
Hi @DeliaBarron, no need to be sorry! Thanks for the changes and being receptive to feedback (: 

The button position looks fine to me, so let's just fix the `col`-related stuff and we should be good. Here's the link to the [Bootstrap layout documentation](https://getbootstrap.com/docs/5.0/layout/columns/) in case you need it!

**Review by domlimm — COMMENTED** [review]
Looks good at my end too 👍🏻 . Have the same comments as @weiquu.

**Comment by DeliaBarron** [comment]
Sure!!! I removed those lines and changed the .col classes. I think the inputs are not shrinking weirdly anymore.
Without lines 24-34 at scss document the 'log-severity-level' form-check wasn't wrapping correctly, therefore the extra changes.

File: src/web/app/pages-monitoring/logs-page/logs-page.component.html [hunk 3]
```diff
@@ -98,7 +98,9 @@ <h1>Logs</h1>
             <label for="source-file" class="fw-bold">Source location file</label>
             <input id="source-file" class="form-control col-12" [(ngModel)]="formModel.filters.sourceLocation!.file">
           </div>
-          <div class="col-12 mt-2">
+        </div>
+        <div class="col-sm-6 col-12">
+          <div class="col-12">
```
Comment (weiquu) [inline src/web/app/pages-monitoring/logs-page/logs-page.component.html:103]: Can revert to `<div class="col-12 mt-2">`

File: src/web/app/pages-monitoring/logs-page/logs-page.component.html [hunk 4]
```diff
@@ -112,7 +114,9 @@ <h1>Logs</h1>
             </select>
           </div>
         </div>
-        <div class="col-5">
+      </div>
+      <div class="row">
```
Comment (weiquu) [inline src/web/app/pages-monitoring/logs-page/logs-page.component.html:118]: Can be deleted

File: src/web/app/pages-monitoring/logs-page/logs-page.component.html [hunk 5]
```diff
@@ -121,7 +125,9 @@ <h1>Logs</h1>
             <label for="regkey" class="fw-bold">User Registration Key</label>
             <input id="regkey" class="form-control" [(ngModel)]="formModel.filters.userInfoParams!.regkey" [disabled]="!isAdmin">
           </div>
-          <div class="col-12 mt-2">
+        </div>
+        <div class="col-sm-6 col-12">
+          <div class="col-12">
```
Comment (weiquu) [inline src/web/app/pages-monitoring/logs-page/logs-page.component.html:130]: Can revert to `<div class="col-12 mt-2">`

**Review by weiquu — CHANGES_REQUESTED** [review]
Hi @DeliaBarron, as mentioned above, let's not make layout changes to the usual desktop view of the input fields unless there's good reason to (and do let me know if there is a reason!)

If you agree, I've pointed out in the comments which places are causing the layout changes and how to get back the usual layout. Please take a look (:

Do also update the snapshots afterwards!

**Comment by DeliaBarron** [comment]
@weiquu Hello! Yes sorry I'll get the layout as it was before all the movements, just letting the button layout as we agreed right (at the bottom so it doesn't overflow)? everything else as it was displayed? the inputs form Trace ID all the way to Exception /Error Class in two columns...
Screen Shot 2023-04-05 at 14 54 41

like this but with the changes on the clear all button at the bottom right? Sorry I want to be super clear I understand. I'm working on it right now.

File: src/test/java/teammates/test/TestProperties.java [hunk 6]
```diff
@@ -39,7 +39,7 @@ private TestProperties() {
                 prop.load(testPropStream);
             }
 
-            IS_SNAPSHOT_UPDATE = Boolean.parseBoolean(prop.getProperty("test.snapshot.update", "false"));
+            IS_SNAPSHOT_UPDATE = Boolean.parseBoolean(prop.getProperty("test.snapshot.update", "true"));
```
Comment (weiquu) [inline src/test/java/teammates/test/TestProperties.java:42]: Let's not change the default value for this! Snapshots should be checked first to ensure changes are expected before running the update. Do revert the changes

**Review by weiquu — CHANGES_REQUESTED** [review]
Thanks @DeliaBarron, the logs page looks good! Do settle the comment below and update the snapshots and we should be good to go (: 

You can update the snapshots by running `npm run test` and pressing `a` to run all test cases. After that, check through the snapshots to make sure the changes are as expected, then press `u` to update them. You can find more details on snapshot testing [here](https://teammates.github.io/teammates/snapshot-testing.html).

**Comment by DeliaBarron** [comment]
@weiquu Ok I understand now! Thanks for explaining I was lost on the snapshot documentation I thought changes on style did need testing. 
Great!!! Hope its all good now! If you don't mind I'll keep bothering with more questions and do more PR for TEAMMATES' project haha I learned a lot! thanks!
I'll stay tuned if something is missing before the merge!

**Review by weiquu — APPROVED** [review]
LGTM!

File: src/web/app/pages-monitoring/logs-page/logs-page.component.html [hunk 7]
```diff
@@ -36,12 +36,12 @@ <h1>Logs</h1>
               </button>
             </div>
           </div>
-          <div id="logs-to-timepicker" class="col-md-4">
+          <div id="logs-to-timepicker" class="col-md-4 mb-2">
```
Comment (domlimm) [inline src/web/app/pages-monitoring/logs-page/logs-page.component.html:39]: Just a nit. Let's follow the `from-timepicker`'s margin too, i.e., It is using `mt-4`, so that it looks consistent 😄 
```suggestion
          <div id="logs-to-timepicker" class="col-md-4 mb-4">
```

Comment (DeliaBarron) [reply src/web/app/pages-monitoring/logs-page/logs-page.component.html:39]: Sorry I don't understand, should I add a `mt-4` instead of `mb-4`?

Comment (domlimm) [reply src/web/app/pages-monitoring/logs-page/logs-page.component.html:39]: Sorry for the confusion. Please add `mb-4`!

**Comment by DeliaBarron** [comment]
Got it now! Thanks! all done!

**Review by domlimm — APPROVED** [review]
LGTM! Thank you for your contribution!

**Comment by weiquu** [comment]
Hello don't forget to update the snapshots! As a general rule, run `npm run test` after making frontend changes to check if there are any snapshots that need updating (:

**Comment by DeliaBarron** [comment]
I'll do! `npm run test` and `npm run lint`. 
Is there something else I'm missing to do?

**Comment by domlimm** [comment]
LGTM, thanks @DeliaBarron!

**Review by zhaojj2209 — APPROVED** [review]
